src/vos/vos_aggregate.c

```diff
@@ -239,6 +239,23 @@ vos_agg_obj(daos_handle_t ih, vos_iter_entry_t *entry,
 	}
 
 	if (agg_param->ap_discard) {
+		if (agg_param->ap_sub_tree_empty) {
+			struct daos_lru_cache	*occ = vos_obj_cache_current();
+			struct vos_container	*cont;
+			struct vos_object	*obj;
+
+			cont = vos_hdl2cont(agg_param->ap_coh);
+			rc = vos_obj_hold(occ, cont, entry->ie_oid,
+					  agg_param->ap_max_epoch, true, true,
```

Comment:
Using ap_max_epoch isn't correct, it's the max epoch in SV tree. You need to use the hi_epc from iterator parameter (like what we used for dkey iterator prep), which means we need to pass in the iterator parameter (or the hi epoch) to vos_agg_obj().